portal/portal-render-impl/impl/src/java/org/sakaiproject/portal/render/portlet/services/SakaiOptionalPortletContainerServices.java

```diff
@@ -233,45 +231,43 @@ private void setupThread(PortletRequest request, boolean doLog)
 
 		String placementId = (String) request
 				.getAttribute("org.sakaiproject.portal.api.PortalService_placementid");
-		// System.out.println("place from getAttribute = "+placementId);
+		log.debug("place from getAttribute = {}", placementId);
 		if (placementId == null)
 		{
-			if (doLog) M_log.info("No Placement found");
+			if (doLog) log.info("No Placement found");
 			return; // We have nothing to work with
 		}
 
 		Session session = SessionManager.getCurrentSession();
-		// System.out.println("Session = "+session);
+		log.debug("Session = {}", session);
 		if (session == null)
 		{
-			if (doLog) M_log.info("No Session found placementId=" + placementId);
+			if (doLog) log.info("No Session found placementId=" + placementId);
 			return; // We have nothing to work with
 		}
 
-		// System.out.println("UserId="+session.getUserId()+"
-		// UserEID="+session.getUserEid());
+		log.debug("UserId={} UserEID={}", session.getUserId(), session.getUserEid());
 
 		// Check to see if there is already a placement in place
 		Placement ppp = (Placement) ThreadLocalManager.get(CURRENT_PLACEMENT);
-		// System.out.println("ThreadLocal CURRENT_PLACEMENT="+ppp);
+        if (log.isDebugEnabled()) log.debug("ThreadLocal CURRENT_PLACEMENT={}", ppp);
 		if (ppp != null)
 		{
-			// System.out.println("ThreadLocal CURRENT_PLACEMENT
-			// ID="+ppp.getId());
+			log.debug("ThreadLocal CURRENT_PLACEMENT ID={}", ppp.getId());
 			if (placementId.equals(ppp.getId()))
 			{
-				// System.out.println("Thread already setup");
+				log.debug("Thread already setup");
 				return; // Placement in place
 			}
 		}
 
 		// find the tool from some site (ToolConfiguration extends Placement)
 		ToolConfiguration siteTool = SiteService.findTool(placementId);
-		// System.out.println("siteTool="+siteTool);
+		if (log.isDebugEnabled()) log.debug("siteTool={}", siteTool);
```

Comment:
I thought toString on siteTool may be costly, so I protected it. Whatever happens toString will be called and loads of concat could be happening in there. Is that not right?